mlflow/deployments/plugin_manager.py

```diff
@@ -0,0 +1,105 @@
+import warnings
+import sys
+import abc
+
+import entrypoints
+from mlflow.deployments.base_plugin import BasePlugin
+from mlflow.exceptions import MlflowException
+from mlflow.protos.databricks_pb2 import RESOURCE_DOES_NOT_EXIST, INTERNAL_ERROR
+
+
+if sys.version_info >= (3, 4):
```

Comment:
Should be able to remove this check now that we've dropped Python 2 support in #2534 :)

Reply:
Awesome, What is the minimum version we are supporting? 3.5?

Reply:
yep!

Reply:
I just figured pandas officially supports only 3.6+ and mlflow has a dependency on pandas. So shouldn't the minimal supported version must be 3.6?

Reply:
Ah, yeah new releases of pandas only support 3.6 and above, people can still use MLflow with older versions of pandas on Python 3.5.2 (which will [reach EOL in September this year](https://devguide.python.org/#status-of-python-branches))

Reply:
Alright, make sense. But setup.py will always tries to install the lates pandas now which will raise an error for those who has 3.5